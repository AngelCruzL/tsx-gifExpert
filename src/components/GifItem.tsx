import React from 'react';
import { CustomGif } from '../interfaces/CustomGif';

function GifItem({ title, url }: CustomGif) {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
}

export default GifItem;
