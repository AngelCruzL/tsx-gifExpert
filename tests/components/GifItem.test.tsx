import React from 'react';
import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';

import GifItem from '../../src/components/GifItem';

describe('GifItem', () => {
  const title = 'Berserk';
  const url = 'https://media2.giphy.com/media/oNJ3am00JCroA/giphy.gif';

  test('should render', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toBeTruthy();
  });

  test('should match snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('should render the image with the correct src and alt', () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole('img') as HTMLImageElement;
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('should render the title', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
