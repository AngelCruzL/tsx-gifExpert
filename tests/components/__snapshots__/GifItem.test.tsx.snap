// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifItem > should match snapshot 1`] = `
<div>
  <div
    class="card animate__animated animate__fadeIn"
  >
    <img
      alt="Berserk"
      src="https://media2.giphy.com/media/oNJ3am00JCroA/giphy.gif"
    />
    <p>
      Berserk
    </p>
  </div>
</div>
`;
